src/mediator.ts

```diff
@@ -1,19 +1,28 @@
 import { BaseMediator } from "./baseMediator";
+import { IMediatorMiddleware } from "./interfaces";
 
 export class Mediator extends BaseMediator {
-    public Send(command: string, payload: any) {
+    private middlewares: IMediatorMiddleware[] = [];
+
+    public Send(command: string, payload: any): any {
         return this.Process(command, payload);
     }
 
-    public Request(query: string, payload: any) {
+    public Request(query: string, payload: any): any {
         return this.Process(query, payload);
     }
 
-    private Process(msg: string, payload: any) {
+    public RegisterMiddleware(middleware: IMediatorMiddleware): void {
```

Comment:
I would prefer `Use(middleware: IMediatorMiddleware)` rather than `RegisterMiddleware`, I think it's more neat!